File: frontend/types/stock.ts
export interface Stock {
  symbol: string;
  name: string;
  sector: string | null;
  industry: string | null;
  market_cap: number | null;
  float_shares: number | null;
  is_adr: boolean;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface StockPrice {
  id: number;
  symbol: string;
  date: string;
  open: number;
  high: number;
  low: number;
  close: number;
  volume: number;
  vwap: number | null;
}

export interface StockMetrics {
  id: number;
  symbol: string;
  date: string;
  ema20: number | null;
  ema50: number | null;
  ema200: number | null;
  rsi: number | null;
  relative_strength_spy: number | null;
  relative_strength_qqq: number | null;
  distance_to_ema20: number | null;
  distance_to_ema50: number | null;
  distance_to_high_52w: number | null;
  avg_volume_20d: number | null;
  relative_volume: number | null;
  // Additional indicators
  sma50: number | null;
  sma150: number | null;
  sma200: number | null;
  perf_1y: number | null;
  perf_1w: number | null;
  low_52w: number | null;
  adr_percent: number | null;
  avg_volume_10d: number | null;
  current_price: number | null;
}

File: backend/app/schemas/stock.py
from pydantic import BaseModel, Field
from datetime import datetime
from typing import Optional


class StockBase(BaseModel):
    symbol: str = Field(..., max_length=10)
    name: str
    sector: Optional[str] = None
    industry: Optional[str] = None
    market_cap: Optional[float] = None
    float_shares: Optional[float] = None
    is_adr: bool = False


class StockCreate(StockBase):
    pass


class Stock(StockBase):
    is_active: bool = True
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True


class StockPriceBase(BaseModel):
    symbol: str = Field(..., max_length=10)
    date: str
    open: float
    high: float
    low: float
    close: float
    volume: int
    vwap: Optional[float] = None


class StockPrice(StockPriceBase):
    id: int
    created_at: datetime

    class Config:
        from_attributes = True


class StockMetricsBase(BaseModel):
    symbol: str = Field(..., max_length=10)
    date: str
    ema20: Optional[float] = None
    ema50: Optional[float] = None
    ema200: Optional[float] = None
    rsi: Optional[float] = None
    relative_strength_spy: Optional[float] = None
    relative_strength_qqq: Optional[float] = None
    distance_to_ema20: Optional[float] = None
    distance_to_ema50: Optional[float] = None
    distance_to_high_52w: Optional[float] = None
    avg_volume_20d: Optional[int] = None
    relative_volume: Optional[float] = None


class StockMetrics(StockMetricsBase):
    id: int
    created_at: datetime
    updated_at: datetime

    class Config:
        from_attributes = True
